### PR TITLE
chore: move to hatchling

### PR DESCRIPTION
**Pull request by henryiii**
Moving over to using pypa/hatch's hatching.

Removed the `[mypy]` extra as it wasn't maintained and wasn't correct.

### CODE REVIEW

**Comment by henryiii** [comment]
Sorry! I tried to add that last commit via the web editor (the WebAssembly one you get by pressing the period key) and it pushed it to master instead. Reverted. :flushed:

**Comment by henryiii** [comment]
Does Cirrus have isolated builds disabled? Not sure why it's not picking up hatchling. It could be added - but why doesn't it pick it up?

File: .cirrus.yml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 run_tests: &RUN_TESTS
   install_cibuildwheel_script:
-    - python -m pip install -e ".[dev]" pytest-custom-exit-code
+    - pip install hatchling
```
Comment (henryiii) [inline .cirrus.yml:3]:
```suggestion
    - python -m pip --version
    - python -m pip install hatchling
```

**Comment by henryiii** [comment]
Something seems broken with homebrew * isolated installs on M1. Vaguely thought I remember hearing about that somewhere else.

**Comment by joerick** [comment]
Yeah I can recreate that locally, too. Seems it only happens on homebrew python.

https://gist.github.com/joerick/42e9e31ceb5d1dfe43c69fb6a59436e2#file-gistfile1-txt-L5541-L5551

**Comment by ofek** [comment]
There are conflicts btw

**Comment by ofek** [comment]
Maybe one more try? 😅

**Comment by ofek** [comment]
Strange job timeout

**Comment by joerick** [comment]
Checks looks okay now, perhaps a retry fixed it. The other thing I think we'd need is an update to `bin/bump_version.py`, as it has references to setup.cfg. I'll push that now.

**Comment by mayeut** [comment]
I cancelled the travis ci job (which would have failed on windows anyway)

**Comment by henryiii** [comment]
> I cancelled the travis ci job (which would have failed on windows anyway)

Is that an issue with this?

**Comment by mayeut** [comment]
> Is that an issue with this?

No, Windows has been failing the last couple builds.

**Comment by henryiii** [comment]
Any objections to merging?

**Pull request merged**